collect_app/src/main/java/org/odk/collect/android/upload/InstanceServerUploader.java

```diff
@@ -205,8 +205,8 @@ public String uploadOneSubmission(Instance instance, String urlString) throws Up
                     if (messageParser.isValid()) {
                         exception = new UploadException(FAIL + messageParser.getMessageResponse());
                     } else {
-                        exception = new UploadException(FAIL + postResult.getReasonPhrase()
-                                + " (" + responseCode + ") at " + urlString);
+                        Timber.i(FAIL + postResult.getReasonPhrase() + " (" + responseCode + ") at " + urlString);
+                        exception = new UploadException("Failed to upload. Please make sure the form is configured to accept submissions on the server");
```

Comment:
I'd imagine there are other errors that can happen here. Would you be able to investigate and see if you can find how to "detect" that we are seeing a "submission closed" error? Potentially the HTTP response code is a way for us to make that decision. Once we have that I think we'd want to have a test that for the decision in place. Happy to chat about the best way to do that 😁

Reply:
Well, so a `400` always tells us submission closed in this case ?

Reply:
Hmm I don't think so as it's just the generic [Bad Request status](https://www.flickr.com/photos/girliemac/6540669737/in/album-72157628409467125/). Maybe something more interesting comes back in the message body?

Reply:
Will check that! :+1:

Reply:
So, this is what is coming in the body : 
```html
<body>
    <h1>HTTP Status 400 – Bad Request</h1>
    <hr class="line" />
    <p><b>Type</b> Status Report</p>
    <p><b>Message</b> Submissions have been disallowed on this form</p>
    <p><b>Description</b> The server cannot or will not process the request due to something that is perceived to be a client error (e.g., malformed request syntax, invalid request message framing, or deceptive request routing).</p>
    <hr class="line" />
    <h3>Apache Tomcat/8.5.39 (Ubuntu)</h3></body>
```

Reply:
### What way would you suggest other than this?
![image](https://user-images.githubusercontent.com/37476886/78577838-4ba25980-784c-11ea-93fb-f69a4b637045.png)

Reply:
Hmmm it's surprising to me that Aggregate errors don't come back in an `OpenRosaResponse` format (Open Rosa is the standard for the different servers that Collect can speak to) but looking at the [docs](https://docs.opendatakit.org/openrosa-form-submission/#form-submission) I'm not sure that's actually expected. 

@lognaturel are you able to chime in here? Has there been previous though about how to support errors coming back from Open Rosa servers that are more specific than HTTP status codes or is it generally uncharted waters? If so maybe we should be using a more specific response code in Aggregate here (401/403 for example) and then have Collect have error messages closer to the errors [here](https://docs.opendatakit.org/openrosa-form-submission/#server-status-codes).

Reply:
What can I possibly do now ?

Reply:
Sorry @rajkumaar23. There's a very small team of people working on all the ODK tools and we're mostly in different timezones so discussions can move pretty slowly! If you'd like to look at some other issues to contribute to I'd check the ["good first issue"](https://github.com/opendatakit/collect/labels/good%20first%20issue) tag. If you do want to look at something else let me know and we can close this PR for the moment and move discussion to the issue.

Reply:
@seadowg No problem. Let's wait for @lognaturel to reply :)

Reply:
@lognaturel Could you please help?

Reply:
Thanks for taking this on and apologies for the delay in getting back to you, @rajkumaar23.

The first part of the `if` on line 205 displays an `OpenRosa` response if there is one. I wasn't completely sure that was done here but it is and it's just that Aggregate doesn't send a helpful response. Since Aggregate is no longer in active development, I think we should handle it here.

I propose that we use the message you put in, @rajkumaar23, but that we do it in a branch just for 400. That is, if there's a custom message, show the custom message (existing behavior), else if the response code is 400, use the special text you've added, for any other response code, use the reason phrase (existing behavior).